Sources/SwiftDocCTestUtilities/XCTestCase+TemporaryDirectory.swift

```diff
@@ -0,0 +1,55 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+
+// These helpers methods exist to put temp files for different test executions in different locations when running in Swift CI.
+
+public extension XCTestCase {
+    
+    @available(*, deprecated, message: "Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
+    func NSTemporaryDirectory() -> String {
+        return Foundation.NSTemporaryDirectory()
+    }
+    
+    /// Creates a new temporary directory and returns the URL of that directory.
+    ///
+    /// After the current test method has returned the temporary directory is automatically removed.
+    ///
+    /// - Parameters:
+    ///   - pathComponents: Additional path components to add to the temporary URL.
+    ///   - fileManager: The file manager that will create the directory.
+    /// - Returns: The URL of the newly created directory.
+    func createTemporaryDirectory(pathComponents: String..., fileManager: FileManager = .default) throws -> URL {
+        let bundleParentDir = Bundle(for: Self.self).bundleURL.deletingLastPathComponent()
+        let baseURL = bundleParentDir.appendingPathComponent(name.replacingWhitespaceAndPunctuation(with: "-"))
+        
+        var tempURL = baseURL.appendingPathComponent(ProcessInfo.processInfo.globallyUniqueString)
+        for component in pathComponents {
+            tempURL.appendPathComponent(component)
+        }
+        tempURL.standardize()
+        
+        addTeardownBlock {
+            try? fileManager.removeItem(at: baseURL)
+        }
+        try fileManager.createDirectory(at: tempURL, withIntermediateDirectories: true, attributes: nil)
```

Comment:
Should we consider first creating the baseURL directory so we don't accidentally create more directories than we intend here?

```suggestion
        try fileManager.createDirectory(at: baseURL, withIntermediateDirectories: false, attributes: nil)
        try fileManager.createDirectory(at: tempURL, withIntermediateDirectories: true, attributes: nil)
```

(If we decide to also add an additional `TemporarySwiftDocCTestDirectory` than I think it would look more like this:)

```suggestion
        try fileManager.createDirectory(at: baseURL.deletingLastPathComponent(), withIntermediateDirectories: false, attributes: nil)
        try fileManager.createDirectory(at: tempURL, withIntermediateDirectories: true, attributes: nil)
```

Reply:
Yes, that sounds like a good idea.

Reply:
Ah, this runs into the same issue that multiple calls in the same test method have the same base URL so only the first call successfully creates the base directory and for the rest it already exist. We could catch the issue that the director already exist and ignore it but I think it's safe enough to assume that the bundle's URL already exist and that creating the base URL in a separate call doesn't provide much value. I will add a test assertion here instead.

Reply:
I added a check if the item exist before creating it so that the baseURL can be created without intermediate directories.